File: ObjectProgrammingJava/projekt po - 2/src/projekt/rosliny/Barszcz.java
package projekt.rosliny;

import projekt.Organizm;
import projekt.Wypisanie;
import projekt.obsluga.Para;
import projekt.swiat.Swiat;
import projekt.zwierzeta.Zwierze;

import java.awt.*;
import java.util.Random;
import java.util.Vector;

public class Barszcz extends Roslina implements Wypisanie {
    public Barszcz(Swiat swiat, int x, int y, boolean czyZyje){
        polozenieX = x;
        polozenieY = y;
        sila = 10;
        inicjatywa = 0;
        kolor = new Color(65, 54, 225);
        oznaczenie = 'b';
        this.swiat = swiat;
        this.czyZyje = czyZyje;
        this.czyMozeRozmnazac = 0;
        wiek = 0;
    }

    @Override
    public void akcja() {
        przygotujOrganizm();
        Random random = new Random();
        int losuj = random.nextInt(100);
        if(losuj>98){
            super.akcja();
        }
        String status;
        Vector<String> kom = swiat.getKomentarz();
        Organizm[][] swiatOrganizmow = swiat.getSwiatOrganizmow();
        Vector<Para> zwierzeta = szukajZwierzecia(polozenieX, polozenieY);
        if (zwierzeta.size() > 0) {
            for (Para wsp : zwierzeta) {
                status = Wypisz() + " ZABIL " + swiatOrganizmow[wsp.getL()][wsp.getR()].Wypisz();
                swiatOrganizmow[wsp.getL()][wsp.getR()].setCzyZyje(false);
                swiatOrganizmow[wsp.getL()][wsp.getR()] = null;
                kom.add(status);
            }
        }
    }

    private Vector<Para> szukajZwierzecia(int x, int y) {
        Organizm[][] swiatOrganizmow = swiat.getSwiatOrganizmow();
        int sizeX = swiat.getSizeX();
        int sizeY = swiat.getSizeY();
        Vector<Para> vect = new Vector<>();
        if (x - 1 >= 0 && y - 1 > 0 && swiatOrganizmow[x - 1][y - 1] instanceof Zwierze) {
            vect.add(new Para(x-1, y-1));
        }
        if (x - 1 >= 0 && swiatOrganizmow[x - 1][y] instanceof Zwierze) {
            vect.add(new Para(x-1,y));
        }
        if (x - 1 >= 0 && y + 1 < sizeY && swiatOrganizmow[x - 1][y + 1] instanceof Zwierze) {
            vect.add(new Para(x-1,y +1));
        }
        if (y - 1 >= 0 && swiatOrganizmow[x][y - 1] instanceof Zwierze) {
            vect.add(new Para(x,y - 1));
        }
        if (y + 1 < sizeY && swiatOrganizmow[x][y + 1] instanceof Zwierze) {
            vect.add(new Para(x,y +1));
        }
        if (x + 1 < sizeX && y - 1 >= 0 && swiatOrganizmow[x + 1][y - 1] instanceof Zwierze) {
            vect.add(new Para(x+1,y - 1));
        }
        if (x + 1 < sizeX && swiatOrganizmow[x + 1][y] instanceof Zwierze) {
            vect.add(new Para(x+1,y));
        }
        if (x + 1 < sizeX && y + 1 < sizeY && swiatOrganizmow[x + 1][y + 1] instanceof Zwierze) {
            vect.add(new Para(x+1,y + 1));
        }
        return vect;
    }

    @Override
    public void kolizja(Organizm atakujacy) {
        String status;
        Vector<String> kom = swiat.getKomentarz();
        Organizm[][] swiatOrganizmow = swiat.getSwiatOrganizmow();
        status = atakujacy.Wypisz() + " ZJADL " + Wypisz() + " I ZGINAL";
        swiatOrganizmow[atakujacy.getPolozenieX()][atakujacy.getPolozenieY()] = null;
        atakujacy.setCzyZyje(false);
        kom.add(status);
    }

    @Override
    public String Wypisz() {
        return "Barszcz sosnowskiego" + getKoncowka();
    }
}

File: ObjectProgrammingPython/Rosliny/barszcz.py
import random

from Obsluga.para import Para
from Rosliny.roslina import Roslina
from Zwierzeta.cyberOwca import CyberOwca
from Zwierzeta.zwierze import Zwierze


class Barszcz(Roslina):
    def __init__(self, Swiat, x, y, czyZyje):
        super(Barszcz, self).__init__(Swiat, x, y, czyZyje)
        self.sila = 10
        self.inicjatywa = 0
        self.kolor = "#8C008C"
        self.oznaczenie = "b"
        self.czyMozeRozmnazac = 0
        self.wiek = 0

    def akcja(self):
        self.przygotujOrganizm()
        losuj = random.randint(0,100)
        if losuj > 98:
            super().akcja()
        status = ""
        kom = self.swiat.getKomentarz()
        swiatOrganizmow = self.swiat.getSwiatOrganizmow()
        zwierzeta = self.szukajZwierzecia(self.polozenieX, self.polozenieY)
        if len(zwierzeta):
            for wsp in zwierzeta:
                status = self.Wypisz() + " ZABIL " + swiatOrganizmow[wsp.getX()][wsp.getY()].Wypisz()
                swiatOrganizmow[wsp.getX()][wsp.getY()].setCzyZyje(False)
                swiatOrganizmow[wsp.getX()][wsp.getY()] = None
                kom.append(status)


    def kolizja(self, atakujacy):
        if type(atakujacy) is not CyberOwca:
            status = ""
            kom = self.swiat.getKomentarz()
            swiatOrganizmow = self.swiat.getSwiatOrganizmow()
            status = atakujacy.Wypisz() + " ZJADL " + self.Wypisz() + " I ZGINAL"
            swiatOrganizmow[atakujacy.getPolozenieX()][atakujacy.getPolozenieY()] = None
            atakujacy.setCzyZyje(False)
            kom.append(status)
        else:
            status = ""
            kom = self.swiat.getKomentarz()
            swiatOrganizmow = self.swiat.getSwiatOrganizmow()
            status = atakujacy.Wypisz() + " ZJADL " + self.Wypisz() + " I BYL ODPORNY"
            swiatOrganizmow[self.polozenieX][self.polozenieY] = swiatOrganizmow[atakujacy.getPolozenieX()][atakujacy.getPolozenieY()]
            swiatOrganizmow[atakujacy.getPolozenieX()][atakujacy.getPolozenieY()] = None
            atakujacy.setPolozenieX(self.polozenieX)
            atakujacy.setPolozenieY(self.polozenieY)
            self.czyZyje = False
            kom.append(status)

    def szukajZwierzecia(self, x, y):
        swiatOrganizmow = self.swiat.getSwiatOrganizmow()
        sizeX = self.swiat.getSizeX()
        sizeY = self.swiat.getSizeY()
        listaPustych = []
        if x - 1 >= 0 and y - 1 >= 0 and type(swiatOrganizmow[x - 1][y - 1]) is Zwierze and type(swiatOrganizmow[x - 1][y - 1]) is not CyberOwca:
            listaPustych.append(Para(x - 1, y - 1))
        if x - 1 >= 0 and type(swiatOrganizmow[x - 1][y]) is Zwierze and type(swiatOrganizmow[x - 1][y - 1]) is not CyberOwca:
            listaPustych.append(Para(x - 1, y))
        if x - 1 >= 0 and y + 1 < sizeY and type(swiatOrganizmow[x - 1][y + 1]) is Zwierze and type(swiatOrganizmow[x - 1][y - 1]) is not CyberOwca:
            listaPustych.append(Para(x - 1, y + 1))
        if y - 1 >= 0 and type(swiatOrganizmow[x][y - 1]) is Zwierze and type(swiatOrganizmow[x - 1][y - 1]) is not CyberOwca:
            listaPustych.append(Para(x, y - 1))
        if y + 1 < sizeY and type(swiatOrganizmow[x][y + 1]) is Zwierze and type(swiatOrganizmow[x - 1][y - 1]) is not CyberOwca:
            listaPustych.append(Para(x, y + 1))
        if x + 1 < sizeX and y - 1 >= 0 and type(swiatOrganizmow[x + 1][y - 1]) is Zwierze and type(swiatOrganizmow[x - 1][y - 1]) is not CyberOwca:
            listaPustych.append(Para(x + 1, y - 1))
        if x + 1 < sizeX and type(swiatOrganizmow[x + 1][y]) is Zwierze and type(swiatOrganizmow[x - 1][y - 1]) is not CyberOwca:
            listaPustych.append(Para(x + 1, y))
        if x + 1 < sizeX and y + 1 < sizeY and type(swiatOrganizmow[x + 1][y + 1]) is Zwierze and type(swiatOrganizmow[x - 1][y - 1]) is not CyberOwca:
            listaPustych.append(Para(x + 1, y + 1))
        return listaPustych

    def Wypisz(self):
        return "Guarana" + self.koncowka()



